src/Network/Algorand/Client.hs

```diff
@@ -22,35 +22,36 @@ import Servant.API.Generic (fromServant)
 import Servant.Client (mkClientEnv, parseBaseUrl, runClientM)
 import Servant.Client.Generic (AsClientT, genericClientHoist)
 
-import qualified Network.Algorand.Node.Api as Api
+import qualified Network.Algorand.Api as Api
 
-import Network.Algorand.Node.Api (Api (..), ApiAny (..), ApiV2, Version (..))
+import Network.Algorand.Api (Api (..), ApiAny (..), ApiIdx2, ApiV2, Version (..))
 
--- | URL of the node to connect to.
-type NodeUrl = Text
+-- | URL of the node or indexer to connect to.
+type Host = Text
 
 -- | Error thrown if 'connect' fails.
-data BadNode = WrongNetwork
+data BadHost = WrongNetwork
   { wnExpected :: Text
   , wnActual :: Text
   }
 
-instance Show BadNode where
+instance Show BadHost where
   show WrongNetwork{wnExpected, wnActual} = mconcat
     [ "Bad node: the node is connected to `" <> T.unpack wnActual <> "`, "
     , "we expected `" <> T.unpack wnExpected <> "`."
     ]
 
-instance Exception BadNode
+instance Exception BadHost
 
-newtype AlgoClient = AlgoClient
-  { getAlgoClient :: forall m' . MonadIO m' => ApiV2 (AsClientT m')
+data AlgoClient = AlgoClient
+  { getAlgoClient  :: forall m' . MonadIO m' => ApiV2 (AsClientT m')
+  , getAlgoIndexer :: forall m' . MonadIO m' => ApiIdx2 (AsClientT m')
   }
 
--- | Connect to a node and make sure it is working on the expected network.
+-- | Connect to a node or indexer and make sure it is working on the expected network.
 connect
   :: forall m. (MonadIO m, MonadThrow m)
-  => NodeUrl
+  => Host
   -- ^ URL of the node.
```

Comment:
APIs are still coupled, aren't they? You pass only one host here, but return `AlgoClient` which provides access to both indexer and node APIs. So how is it supposed to work? If I pass node host here, `getAlgoIndexer` won't work because indexer's host is different.

Reply:
take a look please

also, pay attention for this check
https://github.com/serokell/haskell-algorand-sdk/blob/f6dfc7f08c35cddb8d8ea9993f4d063af12aebd7/src/Network/Algorand/Client.hs#L50-L53

here we request version to know current genesis id, but for new indexer I can't see same version endpoint.. so such behaviour (for both, node and indexer) can be incorrect

Reply:
> here we request version to know current genesis id, but for new indexer I can't see same version endpoint.. so such behaviour (for both, node and indexer) can be incorrect

In this PR let's ignore API changes and assume that the new indexer has exactly the same endpoints as the old one. I've created [SDAO-479](https://issues.serokell.io/issue/SDAO-479) to update APIs accordingly.